src/conda_package_handling/streaming.py

```diff
@@ -0,0 +1,45 @@
+"""
+Exception-compatible adapter from conda_package_streaming.
+"""
+
+from __future__ import annotations
+
+import os
+import os.path
+from tarfile import TarError
+from zipfile import BadZipFile
+
+from conda_package_streaming.extract import exceptions as cps_exceptions
+from conda_package_streaming.extract import extract_stream, package_streaming
+
+from . import exceptions
+
+
+def _extract(fn: str, dest_dir: str, components: list[str]):
```

Comment:
Why is this function protected? There's nothing else going on in this module, so I assume it's being exported and used somewhere else. Doesn't it make sense to advertise it as such?

Also, `fn` is very ambiguous. I'm pretty sure it means `file_name`. Could we please just use that? There's other places in the code too that uses `fn` too.

Reply:
_ is pretty weak but I wanted to discourage this as part of the api. In the docstring it suggests using conda-package-streaming directly instead.